src/inmanta/data/__init__.py

```diff
@@ -449,7 +276,11 @@ def get_value(self, value: object) -> Optional[PRIMITIVE_SQL_TYPES]:
         if isinstance(value, self.base_type):
             # It is as expected
             return value
-        if issubclass(self.base_type, (str, int, bool)) and isinstance(value, (str, int, bool)):
+        if self.base_type == bool:
+            return pydantic.validators.bool_validator(value)
+        if self.base_type == datetime.datetime and isinstance(value, str):
+            return dateutil.parser.isoparse(value)
```

Comment:
If I understand correctly this method is ultimately an API boundary because the typing of the actual enpoints is too generic to do any type conversions or validation. Therefore, I think we should apply any such conversions here, meaning that if this is a naive datetime, we interpret it as UTC. An example scenario where I think this could be the case (and could currently produce incorrect results) is when the user requests a filter for a certain time range.

Reply:
added